jdat_notebooks/ifu_optimal/ifu_optimal.ipynb

```diff
@@ -0,0 +1,1141 @@
+      "  warnings.warn('glue.external.echo is deprecated, import from echo directly instead')\n"
+     ]
+    }
+   ],
+   "source": [
+    "import time\n",
+    "\n",
+    "import numpy as np\n",
+    "\n",
+    "import scipy\n",
+    "\n",
+    "import specutils\n",
+    "from specutils import Spectrum1D\n",
+    "\n",
+    "from jdaviz import CubeViz\n",
+    "\n",
+    "from regions import CircleSkyRegion\n",
+    "\n",
+    "from photutils import CircularAperture, SkyCircularAperture, aperture_photometry \n",
+    "\n",
+    "from astropy.io import fits\n",
+    "from astropy import wcs\n",
+    "import astropy.units as u\n",
+    "from astropy.stats import sigma_clip\n",
+    "from astropy.utils.data import download_file"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%matplotlib inline\n",
+    "import matplotlib.pyplot as plt\n",
+    "from matplotlib.colors import LogNorm"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Introduction\n",
+    "\n",
+    "This notebook illustrates various extraction methods for a point source in JWST NIRSpec IFU data. First we\n",
+    "demonstrate a number of regular extraction techniques, including subset extraction with Cubeviz, simple sum over spaxels, cylindrical aperture, and conical aperture photometry. Then we compare optimal extraction using a WebbPSF model PSF to optimal extraction using a reference star PSF. \n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Read in Simulated NIRSpec IFU Cube\n",
+    "\n",
+    "A faint (quasar) point source was simulated using the NIRSpec Instrument Performance Simulator (IPS), then run through the JWST Spec2 pipeline. We will use this for our science dataset.\n",
+    "\n",
+    "We read in the data both with fits.open and Spectrum1D.read, since the cube handling (slicing) we need to do is not implemented in Spectrum1D yet."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Filename: /Users/pogle/.astropy/cache/download/py3/81b1b1aa0989d55f2e8675c6874ee320\n",
+      "No.    Name      Ver    Type      Cards   Dimensions   Format\n",
+      "  0  PRIMARY       1 PrimaryHDU     295   ()      \n",
+      "  1  SCI           1 ImageHDU        64   (33, 39, 2059)   float32   \n",
+      "  2  ERR           1 ImageHDU        12   (33, 39, 2059)   float32   \n",
+      "  3  DQ            1 ImageHDU        12   (33, 39, 2059)   int32 (rescales to uint32)   \n",
+      "  4  WMAP          1 ImageHDU        10   (33, 39, 2059)   float32   \n",
+      "  5  ASDF          1 BinTableHDU     11   1R x 1C   [10315B]   \n",
+      "WCS Keywords\n",
```

Comment:
Outputs need to be cleared out before uploading the notebooks.

Reply:
@robelgeda The Spectrum1D.read bug appears with specutils1.1, but seems to work with 
specutils 1.1.dev326+gc0766f9e .  Should I indicate that specific version in my requirements.txt file?

Reply:
yes please, we are going to move to defining all requirements so please fill in all the versions if possible.